modules/hrm/includes/functions-announcement.php

```diff
@@ -130,3 +130,147 @@ function erp_hr_send_announcement_email( $employee_ids, $post_id ) {
 
     $announcement_email->trigger( $employee_ids, $post_id );
 }
+
+/**
+ * Get all the announcements
+ *
+ * @since 1.9.1
+ *
+ * @param int query arguments
+ *
+ * @return array list of announcements
+ */
+function erp_hr_get_announcements( $args = [] ) {
+    $defaults = [
+        'numberposts'     => 20,
+        'offset'          => 0,
+        'orderby'         => 'post_date',
+        'order'           => 'DESC',
+        'post_type'       => 'erp_hr_announcement',
+    ];
+
+    $args = wp_parse_args( $args, $defaults );
+
+    $announcements = get_posts( $args );
+
+    return $announcements;
+}
+
+/**
+ * Get the count of the announcements
+ *
+ * @since 1.9.1
+ *
+ * @param int query arguments
+ *
+ * @return int count of announcements
+ */
+function erp_hr_get_announcements_count( $args = [] ) {
```

Comment:
For just counting post types, this is inefficient to parse all the post.
Use `wp_count_posts()` method instead.

Reply:
As we can't use date query in `wp_count_posts` and we have to use date query for total filtered announcements so I am keeping this function as is. But changing `erp_hr_get_announcements_status_counts` function to use `wp_count_posts`.